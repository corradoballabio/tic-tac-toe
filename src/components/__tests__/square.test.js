import {render, screen, cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'

import Square from '../square'

afterEach(() => {
  cleanup()
})

test('rendering square component', () => {
  render(<Square testId={1} value={'value-1'}/>)

  const squareElement = screen.getByTestId('sq1')

  expect(squareElement).toBeInTheDocument()
  expect(squareElement).toHaveTextContent('value-1')
  expect(squareElement).toContainHTML('button')
})

test('matches square snapshot', () => {
  const tree = renderer.create(<Square testId={1}/>).toJSON()
  expect(tree).toMatchSnapshot()
})