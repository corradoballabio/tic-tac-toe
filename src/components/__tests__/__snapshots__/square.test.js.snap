// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="square undefined"
  data-testid="sq1"
/>
`;

exports[`matches square snapshot 1`] = `
<button
  className="square undefined"
  data-testid="sq1"
/>
`;
